### PR TITLE
Update Black's target-version to py37

### PR DESCRIPTION
**Pull request by hugovk**
### Description

Black's `target-version` was still set to `['py27', 'py36']`. Set it to `[py37]` instead.

Also update Black and other pre-commit hooks and re-format with Black.

### Checklist

This pull request is:

- [ ] A documentation / typographical error fix
	- Good to go, no issue or tests are needed
- [ ] A short code fix
	- please include the issue number, and create an issue if none exists, which
	  must include a complete example of the issue.  one line code fixes without an
	  issue and demonstration will not be accepted.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.   one line code fixes without tests will not be accepted.
- [ ] A new feature implementation
	- please include the issue number, and create an issue if none exists, which must
	  include a complete example of how the feature would look.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.

**Have a nice day!**

### CODE REVIEW

**Comment by zzzeek** [comment]
okey doke, battle of the trailing comma continues, I guess :)   will run it through and should be good, thanks!

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **zzzeek** to try to get revision 8c5759fd6497c426b75c7de50d45913cf07c1b3e of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
New Gerrit review created for change 8c5759fd6497c426b75c7de50d45913cf07c1b3e: https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/3484

File: tox.ini [hunk 1]
```diff
@@ -141,19 +141,10 @@ commands =
...
-      # flake8-rst-docstrings dependency, leaving it here
-      # in case it requires a version pin
-      pydocstyle
-      pygments
-      black==21.5b1
+      pre-commit
 commands =
-     flake8 ./lib/ ./test/ ./examples/ setup.py doc/build/conf.py {posargs}
-     black --check ./lib/ ./test/ ./examples/ setup.py doc/build/conf.py
+     pre-commit run --all-files --show-diff-on-failure flake8
```
Comment (zzzeek) [inline tox.ini:146]: this doesn't modify the .git checkout at all, right?

**Review by zzzeek — CHANGES_REQUESTED** [review]
pep8 is failing with:

would reformat lib/sqlalchemy/engine/processors.py
Oh no! ���� ���� ����
1 file would be reformatted, 593 files would be left unchanged.

can you fix?  thanks

File: tox.ini (hunk 1 above)
Comment (hugovk) [reply tox.ini:146]: The Black one can, I'll add back `--check` to match the earlier behaviour.

Comment (hugovk) [reply tox.ini:146]: Ah wait we can't do that, the Black specific argument would need to go in `.pre-commit-config.yaml` and so affect the normal pre-commit use.

Would you prefer I revert the last commit and update the pins in `tox.ini`?

Comment (zzzeek) [reply tox.ini:146]: I think so unless pre-commit has a way to add arguments to individual hooks when you run them on the commandline.

tox has to just be able to run without mutating anything, obviously.

Comment (hugovk) [reply tox.ini]: I personally don't mind using tox to also run formatters :)

Anyway, I've updated the pins in tox.ini instead to keep it consistent.

Comment (zzzeek) [reply tox.ini]: running tox to run formatters is fine but that would be a different tox task here.   thanks for updating!

**Comment by zzzeek** [comment]
i can re-test here once processors.py is adjusted (if needed)

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **CaselIT** to try to get revision b3aedf5570d7e0ba6c354e5989835260d0591b08 of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
Patchset b3aedf5570d7e0ba6c354e5989835260d0591b08 added to existing Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/3484

**Comment by sqla-tester** [comment]
**Federico Caselli** (CaselIT) wrote:

seems all right. we are getting some conflicts, but they should be easy enough to solve

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/3484

**Comment by sqla-tester** [comment]
**mike bayer** (zzzeek) wrote:

it's great.  thanks for the contribution Hugo!

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/3484

**Comment by sqla-tester** [comment]
Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/3484 has been **merged**. Congratulations! :)

**Pull request closed without merging**